aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -915,11 +914,30 @@ void apply_lu_factor(const Tensor& input, const Tensor& pivots, const Tensor& in
   auto n = input.size(-1);
   auto leading_dimension = std::max<int64_t>(1, m);
 
-  for (const auto i : c10::irange(batch_size)) {
-    scalar_t* input_working_ptr = &input_data[i * input_matrix_stride];
-    int* pivots_working_ptr = &pivots_data[i * pivots_stride];
-    int* infos_working_ptr = &infos_data[i];
-    lapackLu<scalar_t>(m, n, input_working_ptr, leading_dimension, pivots_working_ptr, infos_working_ptr);
+  std::function<void(int64_t, int64_t)> loop = [](int64_t, int64_t) {};
+  loop = [&](int64_t start, int64_t end) {
```

Comment:
nit: `const auto loop = [&](...) { ... }` but up to you.

Reply:
Will change

Reply:
Yep, using `std::function` explicitly creates unnecessary overhead over just using a lambda via `auto`